Mu2eG4/src/PhysicalVolumeHelper.cc

```diff
@@ -93,11 +93,11 @@ namespace mu2e {
     // Loop over physical volume store.
     G4PhysicalVolumeStore* pstore = G4PhysicalVolumeStore::GetInstance();
     unsigned current(0u);
-    for ( std::vector<G4VPhysicalVolume*>::const_iterator i=pstore->begin(); i!=pstore->end(); ++i){
+    for (auto i=pstore->begin(); i!=pstore->end(); ++i, ++current){
 
       // Add volume to the map; it's an error if its already there.
       G4VPhysicalVolume* vpv = *i;
-      pair<VolMapType_iterator,bool> ret = _volumeMap.insert( std::make_pair(vpv,_volumeMap.size()) );
+      pair<VolMapType_iterator,bool> ret = _volumeMap.insert( std::make_pair(vpv, current) );
```

Comment:
I would like to check on thing.  In the old code, on the first call to insert, the second argument of make_pair is 0 (_volumeMap.size()).  In the new code, it is 1 ( current ).  Do I have that right?  If so, is that an issue or is numbering arbitrary so long as it matches?

Reply:
Hi Rob,  in the old code the first call is to _volumeMap.size(), because the args to make_pair have to be evaluated before the insert() is called.   So the index was zero in the old code.  While the numbering is arbitrary, the numbers in the two data structures here have to match.

Reply:
Sorry, I misread what you said. In the new code "current" is zero on the initial pass, the increment is executed after the loop.

Reply:
Just to be clear.  You changed the numbering from base 0 to base 1 but that's OK.  Does this have any implications for reading old files?

Reply:
duh.  Thanks.